### PR TITLE
SE-3222 Extend EDXAPP_SITE_CONFIGURATION with default values

### PR DESCRIPTION
**Pull request by s0b0lev**
This PR introduce new merge function for `EDXAPP_SITE_CONFIGURATION` settings. 

**Testing instructions**
This instruction is meant to be run locally:

1. Create a new OpenEDXInstance
2. Open OCIM django shell `make shell`
3. Get created instance: 
```python
>>> instance = OpenEdXInstance.objects.last()
```
4. Create new appserver (without spawning actual deployment, we are care only about app server final configuration)
```python
>>> instance._create_owned_appserver()
```
5. Open UI, check that `Combined ansible vars -> EDXAPP_SITE_CONFIGURATION` looks like this:
```yaml
EDXAPP_SITE_CONFIGURATION:
- values:
    CONTACT_US_CUSTOM_LINK: /contact
```
6. Open instance settings (django admin page `http://localhost:5000/admin/instance/openedxinstance/`)
7. Update `Configuration extra settings`:
```yaml
DEMO_CREATE_STAFF_USER: false
SANDBOX_ENABLE_CERTIFICATES: false
demo_test_users: []
EDXAPP_SITE_CONFIGURATION:
- values:
    CONTACT_US_CUSTOM_LINK: /override-contact-us-link
    TEST_VAR: /test
```
8. Repeat step **4**
9. Open UI, check that `Combined ansible vars -> EDXAPP_SITE_CONFIGURATION` looks like this:
```yaml
EDXAPP_SITE_CONFIGURATION:
- values:
    CONTACT_US_CUSTOM_LINK: /override-contact-us-link
    TEST_VAR: /test
```

**Possible edge scenarios to verify:**
- [ ] Appserver provisioned with default `EDXAPP_SITE_CONFIGURATION`
- [ ] Appserver provisioned with additional `Configuration extra settings`
- [ ] Appserver prodivioned with overridden default  for EDXAPP_SITE_CONFIGURATION (for example: CONTACT_US_CUSTOM_LINK)

**Author notes**
@nizarmah I am sorry for such large instruction, but this is the only way to verify it works. 
Feel free to ask any questions or clarify anything you need for this review. 

Thanks!

### CODE REVIEW

**Review by nizarmah — COMMENTED** [review]
Slight changes needed/nitpicks (sorry in advance for all the trailing comma comments :slightly_frowning_face: )

- [x] I tested this following through the testing instructions
- [x] I read through the code
- [ ] ~~I checked for accessibility issues~~
- [x] Includes documentation
- [ ] ~~I made sure any change in configuration variables is reflected in the corresponding client's `configuration-secure` repository.~~

File: instance/tests/models/test_openedx_site_configuration_mixin.py [hunk 1]
```diff
@@ -128,3 +128,109 @@ def test_html_elements_and_attributes_in_static_content_overrides(self):
...
+        Test that when the static content overrides contain html elements and attributes, there are no errors and the
+        expected ansible SiteConfiguration variables are generated.
+        """
+        instance = OpenEdXInstanceFactory()
+        configuration_extra_settings = {
+            'EDXAPP_SITE_CONFIGURATION': [
+                {
+                    'values': {
+                        'override': True,
+                        'CONTACT_US_CUSTOM_LINK': '/random/place/'
```
Comment (nizarmah) [inline instance/tests/models/test_openedx_site_configuration_mixin.py:143]: Small nit: Trailing comma please.

```suggestion
                        'CONTACT_US_CUSTOM_LINK': '/random/place/',
```

File: instance/tests/models/test_openedx_site_configuration_mixin.py [hunk 2]
```diff
@@ -128,3 +128,109 @@ def test_html_elements_and_attributes_in_static_content_overrides(self):
...
+                    }
```
Comment (nizarmah) [inline instance/tests/models/test_openedx_site_configuration_mixin.py:144]: Small nit: Trailing comma please.

```suggestion
                    },
```

File: instance/tests/models/test_openedx_site_configuration_mixin.py [hunk 3]
```diff
@@ -128,3 +128,109 @@ def test_html_elements_and_attributes_in_static_content_overrides(self):
...
+                }
```
Comment (nizarmah) [inline instance/tests/models/test_openedx_site_configuration_mixin.py:145]: Small nit: Trailing comma please.

```suggestion
                },
```

File: instance/tests/models/test_openedx_site_configuration_mixin.py [hunk 4]
```diff
@@ -128,3 +128,109 @@ def test_html_elements_and_attributes_in_static_content_overrides(self):
...
+            ]
```
Comment (nizarmah) [inline instance/tests/models/test_openedx_site_configuration_mixin.py:146]: Small nit: Trailing comma please.

```suggestion
            ],
```

File: instance/tests/models/test_openedx_site_configuration_mixin.py [hunk 5]
```diff
@@ -128,3 +128,109 @@ def test_html_elements_and_attributes_in_static_content_overrides(self):
...
+        app_server = instance._create_owned_appserver()
+
+        expected_variables = [
+            {
+                'values': {
+                    'override': True,
+                    'CONTACT_US_CUSTOM_LINK': '/random/place/',
+                    'static_template_about_content': 'static_template_about_content',
+                    'homepage_overlay_html': 'homepage_overlay_html',
+                }
```
Comment (nizarmah) [inline instance/tests/models/test_openedx_site_configuration_mixin.py:168]: Small nit: Trailing comma please.

```suggestion
                },
```

File: instance/tests/models/test_openedx_site_configuration_mixin.py [hunk 6]
```diff
@@ -128,3 +128,109 @@ def test_html_elements_and_attributes_in_static_content_overrides(self):
...
+            }
```
Comment (nizarmah) [inline instance/tests/models/test_openedx_site_configuration_mixin.py:169]: Small nit: Trailing comma please.

```suggestion
            },
```

File: instance/tests/models/test_openedx_site_configuration_mixin.py [hunk 7]
```diff
@@ -128,3 +128,109 @@ def test_html_elements_and_attributes_in_static_content_overrides(self):
...
+        """
+        Test that when the static content overrides contain html elements and attributes, there are no errors and the
+        expected ansible SiteConfiguration variables are generated.
+        """
+        instance = OpenEdXInstanceFactory()
+        instance.configuration_extra_settings = yaml.dump({
+            "EDXAPP_SITE_CONFIGURATION": [
+                {
+                    "values": {
+                        "SOME_SETTING": "some value"
```
Comment (nizarmah) [inline instance/tests/models/test_openedx_site_configuration_mixin.py:185]: Small nit: Trailing comma please.

```suggestion
                        "SOME_SETTING": "some value",
```

File: instance/tests/models/test_openedx_site_configuration_mixin.py [hunk 8]
```diff
@@ -128,3 +128,109 @@ def test_html_elements_and_attributes_in_static_content_overrides(self):
...
+                    }
+                },
+                {
+                    "domain": "some-specific-site.com",
+                    "values": {
+                        "SPECIFIC_SETTING": "specific value"
```
Comment (nizarmah) [inline instance/tests/models/test_openedx_site_configuration_mixin.py:191]: Small nit: Trailing comma please.

```suggestion
                        "SPECIFIC_SETTING": "specific value",
```

File: instance/tests/models/test_openedx_site_configuration_mixin.py [hunk 9]
```diff
@@ -128,3 +128,109 @@ def test_html_elements_and_attributes_in_static_content_overrides(self):
...
+                    }
+                },
+                {
+                    "site_id": 3,
+                    "values": {
+                        "CONTACT_US_CUSTOM_LINK": "/custom-contact"
```
Comment (nizarmah) [inline instance/tests/models/test_openedx_site_configuration_mixin.py:197]: Small nit: Trailing comma please.

```suggestion
                        "CONTACT_US_CUSTOM_LINK": "/custom-contact",
```

File: instance/tests/models/test_openedx_site_configuration_mixin.py [hunk 10]
```diff
@@ -128,3 +128,109 @@ def test_html_elements_and_attributes_in_static_content_overrides(self):
...
+        Test that when the static content overrides contain html elements and attributes, there are no errors and the
+        expected ansible SiteConfiguration variables are generated.
+        """
+        instance = OpenEdXInstanceFactory()
+        instance.configuration_extra_settings = yaml.dump({
+            "EDXAPP_SITE_CONFIGURATION": [
+                {
+                    "values": {
+                        "SOME_SETTING": "some value"
+                    }
```
Comment (nizarmah) [inline instance/tests/models/test_openedx_site_configuration_mixin.py:186]: Small nit: Trailing comma please.

```suggestion
                    },
```

File: instance/tests/models/test_openedx_site_configuration_mixin.py [hunk 11]
```diff
@@ -128,3 +128,109 @@ def test_html_elements_and_attributes_in_static_content_overrides(self):
...
+                },
+                {
+                    "domain": "some-specific-site.com",
+                    "values": {
+                        "SPECIFIC_SETTING": "specific value"
+                    }
```
Comment (nizarmah) [inline instance/tests/models/test_openedx_site_configuration_mixin.py:192]: Small nit: Trailing comma please.

```suggestion
                    },
```

File: instance/tests/models/test_openedx_site_configuration_mixin.py [hunk 12]
```diff
@@ -128,3 +128,109 @@ def test_html_elements_and_attributes_in_static_content_overrides(self):
...
+                },
+                {
+                    "site_id": 3,
+                    "values": {
+                        "CONTACT_US_CUSTOM_LINK": "/custom-contact"
+                    }
```
Comment (nizarmah) [inline instance/tests/models/test_openedx_site_configuration_mixin.py:198]: Small nit: Trailing comma please.

```suggestion
                },
```

File: instance/tests/models/test_openedx_site_configuration_mixin.py [hunk 13]
```diff
@@ -128,3 +128,109 @@ def test_html_elements_and_attributes_in_static_content_overrides(self):
...
+                }
```
Comment (nizarmah) [inline instance/tests/models/test_openedx_site_configuration_mixin.py:199]: Small nit: Trailing comma please.

```suggestion
                },
```

File: instance/tests/models/test_openedx_site_configuration_mixin.py [hunk 14]
```diff
@@ -128,3 +128,109 @@ def test_html_elements_and_attributes_in_static_content_overrides(self):
...
+            ]
```
Comment (nizarmah) [inline instance/tests/models/test_openedx_site_configuration_mixin.py:200]: Small nit: Trailing comma please.

```suggestion
            ],
```

File: instance/tests/models/test_openedx_site_configuration_mixin.py [hunk 15]
```diff
@@ -128,3 +128,109 @@ def test_html_elements_and_attributes_in_static_content_overrides(self):
...
+        }
+        app_server = instance._create_owned_appserver()
+        expected_variables = [
+            {
+                "values": {
+                    "SOME_SETTING": "some value",
+                    "CONTACT_US_CUSTOM_LINK": "/contact",
+                    'static_template_about_content': 'TEST override!',
+                    'homepage_overlay_html': 'Text override!',
+                }
```
Comment (nizarmah) [inline instance/tests/models/test_openedx_site_configuration_mixin.py:215]: Small nit: Trailing comma please.

```suggestion
                },
```

File: instance/tests/models/test_openedx_site_configuration_mixin.py [hunk 16]
```diff
@@ -128,3 +128,109 @@ def test_html_elements_and_attributes_in_static_content_overrides(self):
...
+            },
+            {
+                "domain": "some-specific-site.com",
+                "values": {
+                    "SPECIFIC_SETTING": "specific value",
+                    "CONTACT_US_CUSTOM_LINK": "/contact",
+                    'static_template_about_content': 'TEST override!',
+                    'homepage_overlay_html': 'Text override!',
+                }
```
Comment (nizarmah) [inline instance/tests/models/test_openedx_site_configuration_mixin.py:224]: Small nit: Trailing comma please.

```suggestion
                },
```

File: instance/tests/models/test_openedx_site_configuration_mixin.py [hunk 17]
```diff
@@ -128,3 +128,109 @@ def test_html_elements_and_attributes_in_static_content_overrides(self):
...
+            },
+            {
+                "site_id": 3,
+                "values": {
+                    "CONTACT_US_CUSTOM_LINK": "/custom-contact",
+                    'static_template_about_content': 'TEST override!',
+                    'homepage_overlay_html': 'Text override!',
+                }
```
Comment (nizarmah) [inline instance/tests/models/test_openedx_site_configuration_mixin.py:232]: Small nit: Trailing comma please.

```suggestion
                },
```

File: instance/tests/models/test_openedx_site_configuration_mixin.py [hunk 18]
```diff
@@ -128,3 +128,109 @@ def test_html_elements_and_attributes_in_static_content_overrides(self):
...
+            }
```
Comment (nizarmah) [inline instance/tests/models/test_openedx_site_configuration_mixin.py:233]: Small nit: Trailing comma please.

```suggestion
            },
```

**Comment by s0b0lev** [comment]
@nizarmah thanks for review :) 

Switching from ts to python make a bit mess with comma in the end. I will update PR.

**Review by nizarmah — APPROVED** [review]
:+1:

- [x] I tested this using mentioned testing instructions
- [x] I read through the code
- [ ] ~~I checked for accessibility issues~~
- [x] Includes documentation
- [ ] ~~I made sure any change in configuration variables is reflected in the corresponding client's `configuration-secure` repository.~~

**Pull request merged**